src/capellambse/helpers.py

```diff
@@ -578,6 +659,118 @@ def cb(el: etree._Element) -> None:
     return process_html_fragments(markup, cb)
 
 
+def _get_handler(
+    resources: cabc.Mapping[str, fh.FileHandler], name: str, /
+) -> fh.FileHandler | None:
+    return (
+        resources.get(name)
+        or resources.get(name.removesuffix(".team"))
+        or resources.get("\x00")
+    )
+
+
+def embed_images(
+    markup: str,
+    /,
+    resources: cabc.Mapping[str, fh.FileHandler],
+) -> markupsafe.Markup:
+    """Embed images into description text.
+
+    This function replaces ``<img>`` tags linking to an image file in
+    one of the given resources with a base64-encoded data URI containing
+    the image data directly. The original path is preserved in the
+    ``data-capella-path`` attribute.
+
+    See Also
+    --------
+    unembed_images : The inverse operation.
+    """
+
+    def cb(el: etree._Element) -> None:
+        if el.tag != "img":
+            return
+
+        src = el.get("src")
+        if src is None or "://" in src:
+            return
+
+        hdlname, *srcparts = pathlib.PurePosixPath(src).parts
+        srcpath = pathlib.PurePosixPath(*srcparts)
+        hdl = _get_handler(resources, hdlname)
+        if hdl is None:
+            return
+
+        try:
+            content = hdl.read_file(srcpath)
+        except FileNotFoundError:
+            return
+
+        b64content = base64.standard_b64encode(content).decode("ascii")
+        mime, _ = mimetypes.guess_type(srcpath, strict=False)
+        el.set("src", f"data:{mime};base64,{b64content}")
+        el.set("data-capella-path", src)
```

Comment:
```suggestion
    def cb(el: etree._Element) -> None:
        if el.tag != "img":
            return
        src = el.get("src") or ""
        resolved = _get_handler_and_relpath(src, resources)
        if not resolved:
            return
        hdl, relpath = resolved
        try:
            content = hdl.read_file(relpath)
        except FileNotFoundError:
            LOGGER.debug("Image not found: %s", relpath)
            return
        mime, _ = mimetypes.guess_type(relpath.as_posix(), strict=False)
        data_uri = _bytes_to_data_uri(
            content, mime or "application/octet-stream"
        )
        el.set("src", data_uri)
        el.set("data-capella-path", src)
```

Reply:
The `_bytes_to_data_uri` function` which I find actually useful and maybe want to reuse it in other projects too. So maybe it shouldn't have the leading `_`:
```
def _bytes_to_data_uri(data: bytes, mime: str) -> str:
    """Encode raw bytes as a base64 data URI with given MIME type."""
    b64 = base64.standard_b64encode(data).decode("ascii")
    return f"data:{mime};base64,{b64}"
```